vk/backend/backend.go

```diff
@@ -146,6 +149,27 @@ func NewBackend(ctx context.Context, rp runtimeTypes.ContainerRuntimeProvider, p
 		return nil, errors.New("error converting network resource units")
 	}
 
+	kernelVersionRaw, err := os.ReadFile("/proc/sys/kernel/osrelease")
+	if err != nil {
+		kernelVersionRaw = []byte(fmt.Sprintf("problem reading kernel version: %v", err))
+	}
+
+	kernelVersion := strings.TrimSpace(string(kernelVersionRaw))
+
+	// this is actually kind of racy... if mutator runs and upgrades the
+	// package during the window between when this binary starts and this
+	// command runs, we'll render the wrong version. we could plumb through
+	// the version via golang's -ldflags -X, but we'll still have this
+	// problem if we start querying other package versions. best to do this
+	// as early as possible, I think.
+	executorVersionRaw, err := exec.Command("dpkg-query", "--show", "--showformat=${Version}", "titus-executor").CombinedOutput()
```

Comment:
Yea don't do this.

Instead, can you do this?

https://levelup.gitconnected.com/a-better-way-than-ldflags-to-add-a-build-version-to-your-go-binaries-2258ce419d2d

way faster, no exec, much simpler at runtime

Reply:
It is another YAGNI. Start with ldflags and only get fancier if we know we need it.